src/SpacetimeDBClient.cs

```diff
@@ -146,177 +145,192 @@ struct PreProcessedMessage
         private readonly CancellationTokenSource _preProcessCancellationTokenSource = new();
         private CancellationToken _preProcessCancellationToken => _preProcessCancellationTokenSource.Token;
 
+        struct PreProcessContext {
```

Comment:
What is the motivation behind this codeblock change - this code is *extremely* sensitive to changes. Over the last month I've probably spent over 20 hours just looking at the message processing code here and even the smallest change can completely break the SDK.

If we want to make these changes can we do it in a separate PR?

Reply:
@lcodes This PR is to implement protobufecomy, not to refactor a bunch of the code. I would request that you revert the refactoring changes even if the code was working, although in this case it sounds like it's bugged.

If my understanding of the code is correct, then it is primarily refactoring. A change like this can in the best case do nothing and in the worst case introduce a bug right before the release of an Alpha playtest.